lightning/impl/sag_fast.pyx

```diff
@@ -322,14 +322,16 @@ def _sag_fit(self,
                     # compute the norm if not done already
                     for j in range(n_nz):
                         square_norm_X[i] += (data[j] * data[j])
-                a = loss.loss(y_pred - g[i] * square_norm_X[i] / lipschitz, y[i])
-                b = loss.loss(y_pred, y[i]) - 0.5 * square_norm_X[i] * (g[i] * g[i]) / lipschitz
-                if a <= b :
-                    # condition is satisfied, decrease Lipschitz constant
-                    lipschitz /= line_search_scaling
-                else:
-                    # condition not satisfied, decrease step size
-                    lipschitz *= 2.0
+                while True:
```

Comment:
I got NaNs on an exponential loss function. This happened on the first iteration, a possibility which i had not considered.